silkworm/core/protocol/rule_set.hpp

```diff
@@ -65,6 +70,8 @@ class IRuleSet {
     //! \brief See [YP] Section 11.3 "Reward Application".
     //! \param [in] header: Current block to get beneficiary from
     virtual evmc::address get_beneficiary(const BlockHeader& header) = 0;
+
+    virtual BlockReward compute_reward(const Block& block) = 0;
```

Comment:
Please note that a custom `compute_reward` implementation is needed also for `protocol::MergeRuleSet`: before the merge point, it is up to the pre-merge protocol rule set to dictate the behaviour; after that, no reward is assigned by the EL because it is a responsibility of the CL.